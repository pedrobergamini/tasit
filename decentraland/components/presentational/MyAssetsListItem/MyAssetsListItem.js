import React from "react";
import { StyleSheet, View, TouchableHighlight } from "react-native";
import PropTypes from "prop-types";
import {
  responsiveHeight,
  responsiveWidth,
} from "react-native-responsive-dimensions";
import MyAsset from "@presentational/MyAsset";

// Note: Changing to PureComponent for performance boost
// It is possible to still using function component with React.memo HoC
// See more:
// https://reactjs.org/docs/react-api.html#reactpurecomponent
// https://medium.com/groww-engineering/stateless-component-vs-pure-component-d2af88a1200b
export default class MyAssetsListItem extends React.PureComponent {
  render() {
    const { asset, userAction } = this.props;
    return (
      <TouchableHighlight>
        <View style={styles.row}>
          <MyAsset asset={asset} userAction={userAction} />
        </View>
      </TouchableHighlight>
    );
  }
}

MyAssetsListItem.propTypes = {
  asset: PropTypes.object.isRequired,
  userAction: PropTypes.object.isRequired,
};

const styles = StyleSheet.create({
  row: {
    alignItems: "center",
    justifyContent: "center",
    marginBottom: responsiveHeight(1),
    paddingBottom: responsiveHeight(3),
    paddingLeft: responsiveWidth(3),
    paddingRight: responsiveWidth(3),
    paddingTop: responsiveHeight(3),
  },
});
