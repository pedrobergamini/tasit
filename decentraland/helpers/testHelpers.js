import AssetTypes from "@constants/AssetTypes";
const { ESTATE, PARCEL } = AssetTypes;

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: 0,
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: 0,
  priceUSD: 0,
  asset: parcel,
};

export const anAction = {
  getId: async () =>
    "0x1234567890123456789012345678901234567890123456789012345678901234",
};

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
  anAction,
};
