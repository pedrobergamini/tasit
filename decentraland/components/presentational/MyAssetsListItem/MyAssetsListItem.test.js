import React from "react";
import { shallow } from "enzyme";
import MyAssetsListItem from "@presentational/MyAssetsListItem";
import { parcel, parcelUserAction } from "@helpers/testHelpers";

describe("MyAssetsListItem", () => {
  it("renders the component", async () => {
    const asset = parcel;
    const userAction = parcelUserAction;
    expect(
      shallow(<MyAssetsListItem asset={asset} userAction={userAction} />)
    ).toMatchSnapshot();
  });
});
