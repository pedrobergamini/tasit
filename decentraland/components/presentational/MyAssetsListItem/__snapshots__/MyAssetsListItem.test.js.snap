// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyAssetsListItem renders the component 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  underlayColor="black"
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 13.34,
        "paddingBottom": 40.019999999999996,
        "paddingLeft": 22.5,
        "paddingRight": 22.5,
        "paddingTop": 40.019999999999996,
      }
    }
  >
    <MyAsset
      asset={
        Object {
          "id": "0123456789",
          "img": "https://api.decentraland.org/v1/parcels/0/0/map.png",
          "name": "Sample Parcel",
          "type": "PARCEL",
        }
      }
      userAction={
        Object {
          "0x1234567890123456789012345678901234567890": Object {
            "assetId": "0123456789",
            "status": "SUCCESSFUL",
          },
        }
      }
    />
  </Component>
</TouchableHighlight>
`;
